sydent/config/crypto.py

```diff
@@ -46,7 +53,7 @@ def parse_config(self, cfg: "ConfigParser") -> bool:
             print("INFO: Updating signing key format: brace yourselves")
 
             self.signing_key = nacl.signing.SigningKey(
-                signing_key_str, encoder=nacl.encoding.HexEncoder
+                signing_key_str.encode(), encoder=nacl.encoding.HexEncoder
```

Comment:
I wonder if this ought to be `encode("ascii")` so it doesn't fail silently if we give it something exotic?

Reply:
Sounds fair enough. I don't think it makes sense to put Unicode into your key.

Reply:
FWIW I found this `encoder=...` stuff to be fiddley when I was annotating this. But AFAICS they're all meant to take bytes and return bytes. Besides, I think pynacl might want to get rid of them anyway: https://github.com/pyca/pynacl/pull/504#issuecomment-454233998